modules/rules_spring/2.3.2/patches/module_dot_bazel.patch

```diff
@@ -0,0 +1,23 @@
+--- MODULE.bazel
++++ MODULE.bazel
+@@ -1,6 +1,16 @@
+ module(
+-  name = "rules_spring",
+-  version = "2.3.2",
+-  repo_name = "rules_spring",
+-  compatibility_level = 1,
++    name = "rules_spring",
++    version = "2.3.2",
++    compatibility_level = 1,
++    repo_name = "rules_spring",
++)
++
++bazel_dep(name = "rules_python", version = "0.31.0")
```

Comment:
The provided `MODULE.bazel` file in the rules_spring 2.3.2 does simply not work unmodified. ;) - The patch is just a copy of the `MODULE.bazel` provided by https://github.com/salesforce/rules_spring

Reply:
Need to test if this works without modifications...

Reply:
I don't know enough about the Python toolchain setup to assess whether this is safe to add as a non-dev dependency.

Reply:
/cc @rickeylev Can you advise here? How should other Bazel modules register python toolchains usually?

Reply:
short answer: it depends, but usually a library module[1] should just add bazel_dep(rules_python) and nothing else.

long answer: Calling `python.toolchain` merely ensures that version is made available, but it doesn't guarantee it will be used unless the version-binding rules are used. Another module earlier in the BFS (usually the root, but not necessarily) could have requested a different version.

The version-binding rules allow a module to force that their build tools use a version of Python they're known to work with. This allows a module's tool to use one Python version, while other parts of the build can use different Python versions. It also helps reduce the support burden of how many Python versions a tool has to work with. If the version-binding rules are used, then a `python.toolchain()` call is required to ensure that version is available.

[1] a submodule that may show up in arbitrary locations in the module graph

Regardless, a submodule shouldn't set `is_default`. That setting only respected by the root module; it is silently ignored for submodules.

Reply:
It does use rules_python. I will create a new PR with the latest from our repo.